src/docs.js

```diff
@@ -106,16 +109,26 @@ async function documenter(options) {
   if (options.publish) {
     let creds = options.aws;
     if (!creds) {
-      let auth = new client.Auth({
-        credentials: options.credentials,
-        baseUrl: options.authBaseUrl,
-      });
-
-      creds = await auth.awsS3Credentials('read-write', options.bucket, options.project + '/');
+      //Bypassing Tascluster credentials if none are provided in the config file.
+      // For testing purposes only.
+      if (options.credentials['clientId'] === 'bypassTcCreds') {
+        creds = options.credentials;
+      } else {
+        let auth = new client.Auth({
+          credentials: options.credentials,
+          baseUrl: options.authBaseUrl,
+        });
+        creds = await auth.awsS3Credentials(
+          'read-write',
+          options.bucket,
+          options.project + '/'
+        );
+      }
     }
 
     let s3 = new aws.S3(creds.credentials);
-    let s3Stream = S3UploadStream(s3);
+
+    let s3Stream = (options.module || S3UploadStream)(s3);
```

Comment:
I like this, but the identifier `module` is a little vague.. module for what??  How about
```javascript
// allow use of S3UploadStream from the caller (for testing)
let s3Stream = (options.S3UploadStream || S3UploadStream)(s3);
```